src/api/PublicGroupsAPI/index.ts

```diff
@@ -0,0 +1,47 @@
+import { UpdatePublicGroupGoalItem } from "@src/models/UpdatePublicGroupGoalItem";
+import { RetrievePublicGroupGoalItem } from "@src/models/RetrievePublicGroupGoalItem";
+import { getInstallId, createContactRequest } from "@src/utils";
+
+export const getPublicGroupGoals = async (publicGroupParentGoalId: string) => {
+  const url = "https://xcmyippimqii2x76diluvd4ywu0oojhu.lambda-url.eu-west-1.on.aws/";
+  const res = await createContactRequest(url, { method: "getPublicGroupGoals", installId: getInstallId(), publicGroupParentGoalId });
+  return res;
+};
+
+export const joinPublicGroup = async (title: string) => {
+  const url = "https://xcmyippimqii2x76diluvd4ywu0oojhu.lambda-url.eu-west-1.on.aws/";
+  const res = await createContactRequest(url, { method: "joinPublicGroup", installId: getInstallId(), title });
+  return res;
+};
+
+export const quitPublicGroup = async (groupId: string) => {
+  const url = "https://xcmyippimqii2x76diluvd4ywu0oojhu.lambda-url.eu-west-1.on.aws/";
+  const res = await createContactRequest(url, { method: "quitPublicGroup", installId: getInstallId(), groupId });
+  return res;
+};
+
+export const updatePublicGroupGoal = async (publicGroupGoal: UpdatePublicGroupGoalItem) => {
+  const url = "https://xcmyippimqii2x76diluvd4ywu0oojhu.lambda-url.eu-west-1.on.aws/";
+  const res = await createContactRequest(url, { method: "updatePublicGroupGoal", installId: getInstallId(), publicGroupGoal });
+  return res;
+};
+
+export const createRootGroup = async () => {
+  const res = await joinPublicGroup("Root Group"); // creates a new root group
+  return res.response.groups[res.response.groups.length - 1]; // returns the id of the created root group
+};
+
+export const addGroupToRootGroup = async (publicGroupGoal: UpdatePublicGroupGoalItem) => {
+  // eslint-disable-next-line no-param-reassign
+  publicGroupGoal.parentId = await createRootGroup(); // gets the id of the created root group (without recreating another group)
```

Comment:
do not re-assign the value to the function param. let's take a new variable ?

Reply:
I'm not sure exactly how can we do that, 
can you explain more please @Tushar-4781 .